File: 8kyu/(8 kyu) Parse nice int from char problem/(8 kyu) Parse nice int from char problem.ts
// 1 Plain solution
export function get_age(age: string): number {
    const years = age[0];

    return parseInt(years);
}

// 2 Simple solution
export function get_age(age: string): number {
    const years = age.charCodeAt(0) - 48;

    return years;
}

// 3 Optimized solution
export function get_age(age: string): number {
    return +age[0];
}

// 4 Clever solution
export const get_age = (age: string) => +age[0];

// 5 Coding golf
export var get_age = parseInt;

File: 8kyu/(8 kyu) Parse nice int from char problem/(8 kyu) Parse nice int from char problem.py
# 1 Simple solution
def get_age(age):
    years = age[0]
    number = int(years)

    return number

# 2 Optimized solution
def get_age(age):
    years = age[0]

    return int(years)

# 3 Clever solution
def get_age(age):
    return int(age[0])

# 4 Coding golf
get_age=lambda n:int(n[0])
